configuration/pih/scripts/global/zl.js

```diff
@@ -185,32 +185,35 @@ function setUpObsWithObsDateTime(widgetId){
     }
 }
 
-function returnVisitDateValidation(encounterDate,returnVisitDateLessThanEncounterDateMsg,badReturnVisitDateMsg){
+function setupReturnVisitDateValidation(encounterDate,returnVisitDateLessThanEncounterDateMsg,badReturnVisitDateMsg){
 
-   const domEl= jq("#apptDate").find("input[type=text]");
+   const domEl= getField('apptDate.value');
    const yrRange = encounterDate.getFullYear() + ":" + (new Date().getFullYear() + 1);
-   domEl.datepicker('option','yearRange', yrRange); 
-   domEl.prop("readonly", "readonly"); 
-
-  jq(domEl).change(function(){
-    const nextVisitDate = domEl.datepicker('getDate');
-    const months = (nextVisitDate.getFullYear() - encounterDate.getFullYear()) * 12;
-
-    if(months < 12  && nextVisitDate > encounterDate){
-      jq("#return-visit-date-error-message").text('')
-      jq('.submit').prop("disabled", false);
-    }
-    else if(months == 12 && nextVisitDate.getMonth() <= encounterDate.getMonth()){
-      jq("#return-visit-date-error-message").text('')
-      jq('.submit').prop("disabled", false);
-    }else if(months <= 0 && encounterDate > nextVisitDate){
-        jq("#return-visit-date-error-message").text(returnVisitDateLessThanEncounterDateMsg)
-        jq('.submit').prop("disabled", true);
-    }
-    else{
-      jq("#return-visit-date-error-message").text(badReturnVisitDateMsg)
-      jq('.submit').prop("disabled", true);
-    }
-
-  })        
+   domEl.datepicker('option','yearRange', yrRange);
+   domEl.prop("readonly", "readonly");
+
+   const returnVisitDateValidation = function() {
+     const nextVisitDate = domEl.datepicker('getDate');
+     const months = (nextVisitDate.getFullYear() - encounterDate.getFullYear()) * 12;
```

Comment:
This is a "nitpick" and I didn't change it, but I think this variable would be clearer if you *didn't* multiple it by 12 and just called it "differenceInYears" or something like that... it took me a little bit to understand the validation logic below and I think something like this would have been clearer:

`if (differnenceInYears < 1  && nextVisitDate > encounterDate)`

("if the two years are the same, just validate that return visit date is after encounter date'")

Reply:
Everything works fine. I just made a small update.
`const differnenceInYears = nextVisitDate.getFullYear() - encounterDate.getFullYear();`